tests/test_odgovlt.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+from sqlalchemy import create_engine
+from sqlalchemy import MetaData
+from ckanext.harvest.harvesters.ckanharvester import CKANHarvester
+from ckanext.harvest.tests.factories import (HarvestSourceObj, HarvestJobObj,
+                                             HarvestObjectObj)
+import mock_ckan
+import mock
+from ckan import model
+from ckan.tests.helpers import reset_db
+import ckanext.harvest.model as harvest_model
+import json
+import os
+import os.path
+from ckan.tests.factories import Organization
+from ckanext.harvest.tests.lib import run_harvest
+
+
+def was_last_job_considered_error_free():
+    last_job = model.Session.query(harvest_model.HarvestJob) \
+                    .order_by(harvest_model.HarvestJob.created.desc()) \
+                    .first()
+    job = mock.MagicMock()
+    job.source = last_job.source
+    job.id = ''
+    return bool(CKANHarvester._last_error_free_job(job))
+
+
+class TestODGovLt(object):
+    @classmethod
+    def setup(cls):
+        reset_db()
+        harvest_model.setup()
+
+    @classmethod
+    def setUpClass(cls):
+        if os.path.isfile('opendatagov.db'):
+            os.system('rm opendatagov.db')
+        db_url = 'sqlite:///opendatagov.db'
+        con = create_engine(db_url)
+        con.execute('''
+          CREATE TABLE t_rinkmena
+          (ID INTEGER PRIMARY KEY ASC, PAVADINIMAS varchar(250) NOT NULL,\
+SANTRAUKA text, TINKLAPIS text COLLATE BINARY,\
+K_EMAIL text COLLATE BINARY)
+          ''')
+        con.execute('''
+          INSERT INTO t_rinkmena VALUES(1, 'Šilumos tiekimo licencijas\
+ turinčių įmonių sąrašas', '​Šilumos tiekimo\
+ licencijas turinčių įmonių sąrašas',\
+ 'http://www.vkekk.lt/siluma/Puslapiai/licencijavimas/licenciju-turetojai.aspx',\
+'jonaiste.jusionyte@regula.lt')
+          ''')
+        con.execute('''
+          INSERT INTO t_rinkmena VALUES(2, '2014 m. vidutinio\
+ metinio paros eismo intensyvumo duomenys', 'Pateikiama\
+ informacija: kelio numeris, ruožo pradžia, ruožo\
+ pabaiga, vidutinis metinis paros eismo \
+intensyvumas, metai, automobilių tipai',\
+'http://lakd.lrv.lt/lt/atviri-duomenys/vidutinio-m\
+etinio-paros-eismo-intensyvumo-valstybines-reik\
+smes-keliuose-duomenys-2013-m', 'vytautas.timukas@lakd.lt')
+          ''')
+
+    def test_gather(self):
+        db_url = 'sqlite:///opendatagov.db'
+        con = create_engine(db_url)
+        meta = MetaData(bind=con, reflect=True)
+
+        class Tables(object):
+            rinkmena = meta.tables['t_rinkmena']
+
+        clause = Tables.rinkmena.select()
+        opendatagov_database = dict()
+        for row in con.execute(clause):
+            opendatagov_database[row[0]] = row
+        with mock.patch('ckanext.harvest.harvesters.\
+ckanharvester.config') as config:
+            config.__getitem__.side_effect = db_url.split()
```

Comment:
Why do you split `db_url`? `db_url` does not have any spaces, so it will return whole string in a list. If you want to put a string in a list, use `[db_url]` instead.

Reply:
Also, `side_effect` should be a function. If you want to return a value, use `return_value` instead.